### PR TITLE
Synchronization

### LINKED ISSUE #22: Front-end & Back-end: See Other Users - Display Users
#11 

In the See Other Users page, display a list of other users with a Follow button next to them.

### LINKED ISSUE #32: Synchronization: Allow Front-End and Back-End to Interface with each other
Connect Jinja syntax with app routes and supply static data to html references, i.e. pictures.

### LINKED ISSUE #40: Synchronization: HTML edits
#11 #32 Make appropriate edits to the HTML files to allow logins and registration.

### LINKED ISSUE #41: Front-end & Back-end: Follow Users
#11 When clicking the follow button, the picked user is added to the list of followed user for the current user.

### LINKED ISSUE #54: Bug: Extra Commas in userImages.csv
The userImages csv has an issue where some rows have extra commas.

### PR DESCRIPTION
**Pull request by MatteoGisondi**
Closes #22, Closes #32, Closes #40, Closes #41, Closes #54

Unify both frontend and backend to allow user to interface with application.

### CODE REVIEW

**Review by alexbara2000 — APPROVED** [review]
Looks very good and everything works flawlessly

**Comment by alexbara2000** [comment]
Could we add all the issues that were resolved with this merge?

**Review by jongjonathan — APPROVED** [review]
Good changes that leads to working code, looks good

**Review by JosephMezza — APPROVED** [review]
The login, sign up and follow features are now linked to the front end of the website properly. This code can now be merged with the main branch.

File: app.py [hunk 1]
```diff
@@ -66,39 +81,57 @@ def login():
...
 
+
 @app.route('/post')
 def post():
     return render_template('post.html')
 
+
+@app.route('/users' , methods=["GET","POST"])
+def users():
+    usersList = follower.getusers()
```
Comment (LiamBurchellReyes) [inline app.py:126]: Later should be renamed to "getUsers()" for consistency.

File: follower.py [hunk 2]
```diff
@@ -1,98 +1,102 @@
...
+def setListCSV(fileName, listToWrite):
+    with open(fileName, 'w', newline='') as csvfile:
+        writer = csv.writer(csvfile, delimiter=',',quotechar='|', quoting=csv.QUOTE_MINIMAL)
+        writer.writerows(listToWrite)
 
-    for people in followerlist[0]:
-        if people == user:
-           return
+
+def getusers():
```
Comment (LiamBurchellReyes) [inline follower.py:17]: Rename to getUsers() at a later date.

File: follower.py [hunk 3]
```diff
@@ -1,98 +1,102 @@
...
+    userslist.append([user, email, firstName, lastName, password])
+    setListCSV('data/users.csv', userslist)
+
+    imagelist = getListFromCSV('data/userImages.csv')
+    imagelist.append([user])
+    setListCSV('data/userImages.csv', imagelist)
 
-def follow(user, follower):
 
-    followerlist = []
```
Comment (LiamBurchellReyes) [inline follower.py:33]: camelCase. Rename to followerList eventually.

File: follower.py [hunk 4]
```diff
@@ -112,75 +116,71 @@ def getUserFollowers(user):
...
+        imageList += imagesForUser(follower)
+    return imageList
 
-    with open('data/userimages.csv', 'w', newline='') as csvfile:
-        writer = csv.writer(csvfile, delimiter=',',quotechar='|', quoting=csv.QUOTE_MINIMAL)
-        writer.writerows(imagelist)
-        # list with all users and pictures will be transposed in the csv
 
+if __name__ == '__main__':
+    import random
```
Comment (LiamBurchellReyes) [inline follower.py:139]: Should this import be included with the others at the top of the file?

File: templates/main.html [hunk 5]
```diff
@@ -21,43 +23,16 @@
...
-        <img src="static\images\img5.jpg" alt="car5">
-        <img src="static\images\img6.jpg" alt="car6">
-        <img src="static\images\img7.jpg" alt="car7">
-        <img src="static\images\img8.jpg" alt="car8">
-        <img src="static\images\img9.jpg" alt="car9">
+        {% for i in imageList %}
+   <img src = "static/images/{{i}}" alt = "i"/>
+   {% endfor %}
+   {% else %}
+   please login to see pics
```
Comment (LiamBurchellReyes) [inline templates/main.html:34]: #55 To be changed later.

**Review by LiamBurchellReyes — APPROVED** [review]
Future changes to be made. Acceptable for now.

**Pull request merged**